### PR TITLE
AWS Cost Anomalies code

### PR DESCRIPTION
**Pull request by yashbindlish**
*Issue #, if available:*

*Description of changes:* added AWS Cost Anomaly Module

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: static/Cost/300_Optimization_Data_Collection/Code/Optimization_Data_Collector.yaml [hunk 1]
```diff
@@ -18,13 +18,14 @@ Metadata:
...
           - IncludeRightsizingModule
+          - IncludeCostAnomalyModule
           - IncludeInventoryCollectorModule
           - IncludeComputeOptimizerModule
           - ComputeOptimizerRegions
           - IncludeECSChargebackModule
           - IncludeRDSUtilizationModule
           - IncludeOrgDataModule
-          - IncludeBudgetsModule
+         - IncludeBudgetsModule
```
Comment (iakov-aws) [inline static/Cost/300_Optimization_Data_Collection/Code/Optimization_Data_Collector.yaml:28]: typo?

Comment (yashbindlish) [reply static/Cost/300_Optimization_Data_Collection/Code/Optimization_Data_Collector.yaml:28]: Thanks for pointing this out. Will correct this.

**Comment by iakov-aws** [comment]
@gaborschulz-aws  please can you check this ?

File: static/Cost/300_Optimization_Data_Collection/Code/Management.yaml [hunk 2]
```diff
@@ -42,6 +42,7 @@ Resources:
               - Effect: "Allow"
                 Action:
                   - "ce:GetRightsizingRecommendation"
+                  - "ce:GetAnomalies"
```
Comment (iakov-aws) [inline static/Cost/300_Optimization_Data_Collection/Code/Management.yaml:45]: Can you create a dedicated policiy?

Comment (yashbindlish) [reply static/Cost/300_Optimization_Data_Collection/Code/Management.yaml:45]: Any specific reason to create a separate dedicated policy? Just another permission with existing ce:GetRightsizingRecommendation policy

Comment (iakov-aws) [reply static/Cost/300_Optimization_Data_Collection/Code/Management.yaml:45]: Yes, we will split this later to optional policies.

**Review by iakov-aws — APPROVED** [review]

File: static/Cost/300_Optimization_Data_Collection/Code/module-cost-explorer-cost-anomaly.yaml [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+                  }
+                  flattened_results.append(flattened_anomaly)
+              return flattened_results
+
+          def calculate_dates(s3_path):
+              end_date = datetime.now().date()
+              start_date = datetime.now().date() - timedelta(days=90) #Cost anomalies are available for last 90days
+
+              # Check if there are any objects in the S3 bucket
+              objects = boto3.client('s3').list_objects_v2(Bucket=BUCKET, Prefix=s3_path)
```
Comment (yprikhodko) [inline static/Cost/300_Optimization_Data_Collection/Code/module-cost-explorer-cost-anomaly.yaml:196]: You need to use pagination if amount of objects greater than 1000 - https://stackoverflow.com/questions/54314563/how-to-get-more-than-1000-objects-from-s3-by-using-list-objects-v2

Comment (iakov-aws) [reply static/Cost/300_Optimization_Data_Collection/Code/module-cost-explorer-cost-anomaly.yaml:196]: Good catch

**Pull request merged**